packages/atomic/src/components/commerce/atomic-product-image/atomic-product-image.spec.ts

```diff
@@ -0,0 +1,393 @@
+        '[part="product-image"]'
+      ),
+      previousButton: element?.shadowRoot?.querySelector(
+        '[part="previous-button"]'
+      ),
+      nextButton: element?.shadowRoot?.querySelector('[part="next-button"]'),
+      indicators: element?.shadowRoot?.querySelector('[part="indicators"]'),
+      indicator: element?.shadowRoot?.querySelector('[part="indicator"]'),
+      activeIndicator: element?.shadowRoot?.querySelector(
+        '[part="indicator active-indicator"]'
+      ),
+    };
+  };
+
+  it('should render correctly with default properties', async () => {
+    const {element, productImage} = await renderProductImage();
+    expect(element!.field).toBe('ec_thumbnails');
+
+    expect(productImage).toBeInTheDocument();
+    expect(productImage).toHaveAttribute('src', DEFAULT_IMAGE);
+  });
+
+  describe('#previousImage', () => {
+    it('should decrement current image index', async () => {
+      const {element} = await renderProductImage({
+        product: buildFakeProduct({
+          ec_thumbnails: Array(3).fill(DEFAULT_IMAGE),
+        }),
+      });
+
+      // @ts-expect-error private property access for testing
+      expect(element.currentImage).toBe(0);
+
+      await element!.previousImage();
+      // @ts-expect-error private property access for testing
+      expect(element.currentImage).toBe(2);
+
+      await element!.previousImage();
+      // @ts-expect-error private property access for testing
+      expect(element.currentImage).toBe(1);
+    });
+  });
+
+  describe('#nextImage', () => {
+    it('should increment current image index', async () => {
+      const {element} = await renderProductImage({
+        product: buildFakeProduct({
+          ec_thumbnails: Array(3).fill(DEFAULT_IMAGE),
+        }),
+      });
+
+      // @ts-expect-error private property access for testing
+      expect(element.currentImage).toBe(0);
+
+      await element!.nextImage();
+      // @ts-expect-error private property access for testing
+      expect(element.currentImage).toBe(1);
+
+      await element!.nextImage();
+      await element!.nextImage();
+      // @ts-expect-error private property access for testing
+      expect(element.currentImage).toBe(0);
+    });
+  });
+
+  describe('#navigateToImage', () => {
+    it('should navigate to specified image index', async () => {
+      const {element} = await renderProductImage({
+        product: buildFakeProduct({
+          ec_name: 'Test Product',
+          ec_thumbnails: Array(3).fill(DEFAULT_IMAGE),
+        }),
+      });
+
+      // @ts-expect-error private property access for testing
+      expect(element.currentImage).toBe(0);
+
+      await element!.navigateToImage(1);
+      // @ts-expect-error private property access for testing
+      expect(element.currentImage).toBe(1);
+
+      await element!.navigateToImage(2);
+      // @ts-expect-error private property access for testing
+      expect(element.currentImage).toBe(2);
+    });
+  });
```

Comment:
Instead of piercing the veil and accessing private prop, why not have 3 images and check the DOM that you land on the right one?

This would also allow to not mix Act & Asserts.